### PR TITLE
Create module  Forward  in Tmail

### PR DESCRIPTION
**Pull request by ManhNTX**
#836

### CODE REVIEW

File: bitrise.yml [hunk 1]
```diff
@@ -77,6 +77,9 @@ workflows:
             cd ../rule_filter
             flutter pub get && flutter pub run build_runner build --delete-conflicting-outputs
 
+            cd ../forward
+            flutter pub get && flutter pub run build_runner build --delete-conflicting-outputs
+
```
Comment (hoangdat) [inline bitrise.yml:82]: please add it also in `Dockerfile`

File: rule_filter/pubspec.yaml [hunk 2]
```diff
@@ -11,9 +11,6 @@ dependencies:
   flutter:
     sdk: flutter
 
-  model:
-    path: ../model
-
```
Comment (hoangdat) [inline rule_filter/pubspec.yaml:16]: 👍

File: forward/README.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
+<!--
```
Comment (dab246) [inline forward/README.md:1]: change it

File: forward/lib/forward/forward_id.dart [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+enum ForwardIdType {
+  singleton('singleton');
+
+  const ForwardIdType(this.value);
+
+  final String value;
+}
+
+@ForwardIdConverter()
```
Comment (dab246) [inline forward/lib/forward/forward_id.dart:14]: If the object is not marked `JsonSerializable` then `annotation` is not needed

File: forward/lib/forward/forward_id.dart [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+@IdConverter()
```
Comment (dab246) [inline forward/lib/forward/forward_id.dart:15]: same

**Comment by dab246** [comment]
Please write `commit`  in the correct format: `TF-###` or `###-`

File: forward/lib/forward/forward_id.dart [hunk 6]
```diff
@@ -0,0 +1,27 @@
+import 'package:equatable/equatable.dart';
+import 'package:forward/forward/converter/forward_id_coverter.dart';
+import 'package:jmap_dart_client/http/converter/id_converter.dart';
+import 'package:jmap_dart_client/jmap/core/id.dart';
+
+enum ForwardIdType {
```
Comment (dab246) [inline forward/lib/forward/forward_id.dart:6]: IMO, just creating a `factory` for `ForwardId` is enough. No need to create `ForwardIdType` anymore. Ex:
```dart
factory ForwardId.singleton() {
    return ForwardId(Id('singleton'));
}
```

What do you guys mean? @hoangdat @ManhNTX

File: forward/test/forward/get_forward_method_test.dart [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+      final httpClient = HttpClient(dio);
+      final processingInvocation = ProcessingInvocation();
+      final requestBuilder =
+          JmapRequestBuilder(httpClient, processingInvocation);
+      final accountId = AccountId(Id(
+          '29883977c13473ae7cb7678ef767cbfbaffc8a44a6e463d971d23a65c1dc4af6'));
+
+      final getForwardMethod = GetForwardMethod(accountId)
+        ..addIds({ForwardIdSingleton.forwardIdSingleton.id});
+      final getRuleFilterInvocation =
```
Comment (dab246) [inline forward/test/forward/get_forward_method_test.dart:80]: rename it to `getForwardInvocation `

**Comment by hoangdat** [comment]
verified! Test cases of `contact`, `forward` and `rule_filter` were passed

image

**Review by hoangdat — APPROVED** [review]

File: forward/lib/forward/forward_id.dart (hunk 6 above)
Comment (ManhNTX) [reply forward/lib/forward/forward_id.dart:6]: I think create enum type make it clear and easy maintain and edit later

Comment (dab246) [reply forward/lib/forward/forward_id.dart:6]: oke

**Review by dab246 — APPROVED** [review]

**Review by hoangdat — APPROVED** [review]

**Pull request merged**